events/README.md

```diff
@@ -1,42 +1,35 @@
-## The mbed-events library ##
+# The mbed-events library
 
-The mbed-events library provides a flexible queue for scheduling events.
+The mbed-events library provides a flexible queue for scheduling events. It can be used as a normal event loop, or it can be backgrounded on a single hardware timer or even another event loop. It is both thread and IRQ safe, and provides functions for easily composing independent event queues.
+
+The library can act as a drop-in scheduler, provide synchronization between multiple threads, or just act as a mechanism for moving events out of interrupt contexts.
+
+## Example
 
 ``` cpp
 #include "mbed_events.h"
 #include <stdio.h>
 
 int main() {
-    // creates a queue with the default size
+    // Creates a queue with the default size
     EventQueue queue;
 
-    // events are simple callbacks
+    // Events are simple callbacks
     queue.call(printf, "called immediately\n");
     queue.call_in(2000, printf, "called in 2 seconds\n");
     queue.call_every(1000, printf, "called every 1 seconds\n");
 
-    // events are executed by the dispatch method
+    // Events are executed by the dispatch method
     queue.dispatch();
 }
 ```
 
-The mbed-events library can be used as a normal event loop, or it can be
-backgrounded on a single hardware timer or even another event loop. It is
-both thread and irq safe, and provides functions for easily composing 
-independent event queues.
-
-The mbed-events library can act as a drop-in scheduler, provide synchronization
-between multiple threads, or just act as a mechanism for moving events out of 
-interrupt contexts.
-
-### Usage ###
+## Usage
 
-The core of the mbed-events library is the [EventQueue](EventQueue.h) class,
-which represents a single event queue. The `EventQueue::dispatch` function
-runs the queue, providing the context for executing events.
+The core of the mbed-events library is the [EventQueue](https://docs.mbed.com/docs/mbed-os-api/en/mbed-os-5.2/api/classevents_1_1EventQueue.html) class, which represents a single event queue. The `EventQueue::dispatch` function runs the queue, providing the context for executing events:
```

Comment:
Out of curiousity, why the single line? Doesn't markdown join sequential lines? Single lines may make this harder to read/edit in terminals (not uncommon for READMEs).